### DISCUSSION TITLE
Allow Unverified Users to Log In and Resend Verification Email

### DISCUSSION THREAD

**Discussion by cdwmhcc** [directus/directus#24883] [Feature Requests]
### Description:

Currently, during the user registration process, a verification email is sent to the user's provided email address. If the user does not receive this verification email, they are unable to proceed with any further actions within the application.

### Issues:

Email Delivery Failures: Users may not receive verification emails due to various reasons such as spam filters or server issues, leading to frustration and potential abandonment of the registration process.

### Improvement:

To improve user experience and engagement, it is proposed to modify the registration and login process to allow users who have not yet verified their email addresses to log in. This adjustment will provide them with the opportunity to resend the verification email and continue exploring the application and can view restricted profiles.

**Comment by hanneskuettner** [comment]
Resending verification emails is currently already supported, as users with the `unverified` can simply register again. I do agree however, that a "Resend verification email" button would go a long way on the current register page.

As for the ability for unverified users to log in, that would require a proper feature request process as this would certainly require a few more considerations than just allowing them to log in. So please feel free to [open a Feature Request](https://github.com/directus/directus/discussions/new?category=feature-requests)

**Comment by cdwmhcc** [comment]
> that would require a proper feature request process as this would certainly require a few more considerations than just allowing them to log in

Your consideration is correct. For unverified users, the only necessary function is the ability to resend verification emails after logging in. Other interactions with the app seem unnecessary until verification is complete.

**Comment by DanielBiegler** [comment]
> For unverified users, the only necessary function is the ability to resend verification emails

They can do that by registering again.

> after logging in

Ahh, could be a nice to have but thats a new thing. Currently unauthorized users see this screen when logging in:

![image](https://github.com/directus/directus/assets/14810858/c7897e74-025f-4987-85b3-5d3d35cdd8ca)

Are you proposing a new message and a new button here specifically for unverified users?

**Comment by cdwmhcc** [comment]
> Are you proposing a new message and a new button here specifically for unverified users?

#### Current Message and Button:
- Title: No App Access
- Description: This user isn't allowed to use the admin app.
- Button: Switch User

#### Proposed Message and Button:
- Title: Account Not Verified
- Description: Please check your email to complete the verification process. If you didn't receive the email, click below button to resend it.
- Description1: Check your email to verify your account, or resend the verification email.
- Button: Resend Verification Email

<hr />

![image](https://github.com/directus/directus/assets/8030134/8f9a9ae4-9aa1-4e9d-9451-bc105c1a8d60)

<hr />

![image](https://github.com/directus/directus/assets/8030134/fc16036a-c767-4e3e-b705-2b575d67abb1)

**Comment by Killa777hs** [comment]
> > Are you proposing a new message and a new button here specifically for unverified users?
> 
> #### Current Message and Button:
> - Title: No App Access
> - Description: This user isn't allowed to use the admin app.
> - Button: Switch User
> 
> #### Proposed Message and Button:
> - Title: Account Not Verified
> - Description: Please check your email to complete the verification process. If you didn't receive the email, click below button to resend it.
> - Description1: Check your email to verify your account, or resend the verification email.
> - Button: Resend Verification Email
> 
> <hr />
> 
> ![image](https://github.com/directus/directus/assets/8030134/8f9a9ae4-9aa1-4e9d-9451-bc105c1a8d60)
> 
> <hr />
> 
> ![image](https://github.com/directus/directus/assets/8030134/fc16036a-c767-4e3e-b705-2b575d67abb1)
>

**Comment by BorisKamp** [comment]
This is important @Directus!

I think it's quite easy to implement in the core, all we need imo is the following:

1. A resend verification email endpoint
2. Will always return a 200, no matter if the email address is not found (security)
3. Will resend the verification email with a new fresh token of the e-mail address is found.

Im coming from Strapi and LOVE Directus, it's way better and has more to offer. But this is clearly missing. You cannot expect user to re-register again vs resending the verification e-mail.

**Comment by rijkvanzanten** [comment]
@BorisKamp A PR is more than welcome! Especially if it's quite easy to implement 😉
